data/trie/sync.go

```diff
@@ -229,19 +227,21 @@ func trieNode(data interface{}) (node, error) {
 
 func (ts *trieSyncer) requestNodes() uint32 {
 	ts.nodeHashesMutex.Lock()
-	numRequested := uint32(len(ts.nodeHashes))
-	for hash := range ts.nodeHashes {
-		ts.requestHandler.RequestTrieNodes(ts.shardId, []byte(hash), ts.topic)
+	numUnResolvedNodes := uint32(len(ts.nodeHashes))
+	for hash, found := range ts.nodeHashes {
+		if !found {
+			ts.requestHandler.RequestTrieNodes(ts.shardId, []byte(hash), ts.topic)
+		}
 	}
 	ts.nodeHashesMutex.Unlock()
 
-	return numRequested
+	return numUnResolvedNodes
```

Comment:
Should be returned the number of all nodeHashes or only the number of requested ones?

Reply:
it should return all the nodes which are not finished.

Reply:
ok